File: print duplicates in a string.py
# space: O(1)
def fillCharCounts(string, countArr):
    for i in string:
        countArr[ord(i)] += 1
def printDuplicates(S):
    countArr = [0 for i in range(256)]
    fillCharCounts(S, countArr)

    # print character having count more than 0
    for i in range(256):
        if countArr[i] > 1:
            print(f'{chr(i)} : {countArr[i]}')


printDuplicates("test string")

File: print duplicates in a string.cpp
// space is O(1)
void printDuplicates(string &S) {
    int counts[26] = { 0 };

    for (int i = 0; i < S.size(); i++) {
        counts[S[i] - 'a'] += 1;
    }
    for (int i = 0; i < 26; i++) {
        if (counts[i] > 1) {
            cout << (char)(i + 'a') << " " << counts[i] << endl;
        }
    }
}
